crypto/s2n_composite_cipher_aes_sha.c

```diff
@@ -73,23 +74,28 @@ static uint8_t s2n_composite_cipher_aes128_sha_available(void)
 {
     /* EVP_aes_128_cbc_hmac_sha1() returns NULL if the implementations aren't available.
      * See https://github.com/openssl/openssl/blob/master/crypto/evp/e_aes_cbc_hmac_sha1.c#L952
+     *
+     * Composite ciphers cannot be used when FIPS mode is set.
```

Comment:
This is not documented by OpenSSL but is caught by our unit tests. OpenSSL throws the following error when we attempt to use any composite cipher:
```
fips_enc.c(100): OpenSSL internal error, assertion failed: Cipher ctrl previous FIPS forbidden algorithm error ignored
```

This error is based on the absence of the ```EVP_CIPH_FLAG_FIPS``` flag on these composite ciphers. My guess is that the changes made to implement composite ciphers were not certified by NIST for FIPS 140-2 compliance.

I can add another sentence to the comment explaining this, but don't have a useful link to provide.